### PR TITLE
Integrate badge field in project user item

### LINKED ISSUE #2235: Integrate user badge (additional roles)
Use graphql api

### PR DESCRIPTION
**Pull request by samshara**
- fixes  #2235
- depends on https://github.com/the-deep/server/pull/1028

* Use graphQL API for fetch, delete, edit user 
* Show and edit badge (additional permission)
## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [ ] permission checks
- [ ] translations

### CODE REVIEW

File: app/views/ProjectEdit/Users/UserList/index.tsx [hunk 1]
```diff
@@ -105,57 +225,76 @@ function UserList(props: Props) {
             cellRenderer: ActionCell,
             cellRendererParams: (userId, data) => ({
                 itemKey: userId,
-                onEditClick: handleEditMembershipClick,
-                onDeleteClick: triggerMembershipDelete,
+                onEditClick: () => handleEditProjectUserClick(data),
+                onDeleteClick: () => handleRemoveUserFromProject(data.id),
```
Comment (tnagorra) [inline app/views/ProjectEdit/Users/UserList/index.tsx]: We should not create inline functions on cell renderer params

File: app/views/ProjectEdit/Users/UserList/index.tsx [hunk 2]
```diff
@@ -57,44 +143,78 @@ function UserList(props: Props) {
...
+                }
+                const {
+                    deletedResult,
+                } = response.project.projectUserMembershipBulk;
+
+                const [deletedUser] = deletedResult ?? [];
+                alert.show(
+                    `Successfully deleted ${deletedUser?.member.displayName}`,
+                    { variant: 'success' },
+                );
```
Comment (tnagorra) [inline app/views/ProjectEdit/Users/UserList/index.tsx]: We should check if deletion this was successful.
We should also alert if it was not.
Also, deletedUser can be undefined on line 188, which means the deletion failed.
Let's check `ok` to see if there's error

File: app/views/ProjectEdit/Users/UserList/AddUserModal/index.tsx [hunk 3]
```diff
@@ -110,41 +152,89 @@ function AddUserModal(props: Props) {
...
+                    return;
+                }
+                const {
+                    errors,
+                    result,
+                } = response.project.projectUserMembershipBulk;
+
+                const [err] = errors ?? [];
+                const [user] = result ?? [];
+                if (errors && errors[0]) {
```
Comment (tnagorra) [inline app/views/ProjectEdit/Users/UserList/AddUserModal/index.tsx]: Do we need to check for the errors to have an item? can we just not check for length?

File: app/views/ProjectEdit/Users/UserList/AddUserModal/index.tsx [hunk 4]
```diff
@@ -110,41 +152,89 @@ function AddUserModal(props: Props) {
...
+                    const formError = transformToFormError(removeNull(err) as ObjectError[]);
+                    setError(formError);
+                } else {
+                    alert.show(
+                        projectUserToEdit
+                            ? `Successfully updated ${user?.member.displayName}`
+                            : `Successfully added ${user?.member.displayName}`,
+                        { variant: 'success' },
```
Comment (tnagorra) [inline app/views/ProjectEdit/Users/UserList/AddUserModal/index.tsx]: What happens if there is no user?

File: app/types/project.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import { BasicElement } from '.';
 import { OrganizationDetails } from './organization';
 
+type badges = 'QA';
```
Comment (tnagorra) [inline app/types/project.tsx]: We should get this enum from generated/types.tsx

**Review by tnagorra — CHANGES_REQUESTED** [review]

**Review by subinasr — APPROVED** [review]

**Pull request merged**